client/HubApi.ts

```diff
@@ -124,11 +124,16 @@ export default class HubApi<DB extends BehaviorType = BehaviorType.POPUP> { // D
         return this._request(requestBehavior, RequestType.MANAGE_CASHLINK, [request]);
     }
 
-    public checkout<B extends BehaviorType = DB>(
-        request: Promise<CheckoutRequest> | CheckoutRequest,
+    public checkout<R extends CheckoutRequest, B extends BehaviorType = DB>(
+        request: Promise<R> | R,
         requestBehavior: RequestBehavior<B> = this._checkoutDefaultBehavior as any,
-    ): Promise<B extends BehaviorType.REDIRECT ? void : SignedTransaction> {
-        return this._request(requestBehavior, RequestType.CHECKOUT, [request]);
+    ): Promise<B extends BehaviorType.REDIRECT
+        ? void
+        : R extends { version: 2 }
+            ? SimpleResult | SignedTransaction
+            : SignedTransaction // Note: this will not be guaranteed anymore when checkout supports external nim wallets
+    > {
+        return this._request(requestBehavior, RequestType.CHECKOUT, [request]) as any;
```

Comment:
Is `as any` necessary now or is it a leftover from experimenting?

Reply:
It's required unfortunately because `ResultByRequestType` distringuishes the result only by request type, not request, and therefore gives `SimpleResult | SignedTransaction`. But `SimpleResult` can not be mapped to `SignedTransaction`.